### ISSUE TITLE
[Dropdown] “search selection” ones are selecting on lost focus

### ISSUE THREAD

**Issue by daltones** [Semantic-Org/Semantic-UI#4468]
Search selection dropdowns are selecting the first option right after it looses focus (eg: by clicking outside). It affects multiple search selections too.

It can be seen in [examples page](http://semantic-ui.com/modules/dropdown.html#/examples).

Note that if you just focus the dropdown and tap <kbd>Esc</kbd> to dismiss the menu, even this way an option is selected when it looses focus.

**Comment by ismay** [comment]
I have exactly the same problem.

**Comment by daltones** [comment]
This problem appeared in v2.2.3.

But I think there's a more general problem even in previous releases:
- In v2.2.3, first suggested option is selected on lost focus even if you type nothing and press <kbd>Esc</kbd> (most annoying issue).
- In previous versions, empty inputs followed by <kbd>Esc</kbd> do not select any option. But if you type `a` (eg), some suggestions appear and you press <kbd>Esc</kbd> to dismiss the menu, the first option *will be selected* on lost focus. It should just drop the input (or create a new option if `allowAdditions` is set).

**Comment by ismay** [comment]
> This problem appeared in v2.2.3.

Exactly, downgrading to 2.2.1 solves the extra selection problem.

> In previous versions, empty inputs followed by Esc do not select any option. But if you type a (eg), some suggestions appear and you press Esc to dismiss the menu, the first option will be selected on lost focus. It should just drop the input (or create a new option if allowAdditions is set).

Can confirm. Does this for me in 2.2.1

**Comment by dorsha** [comment]
It happens to me as well after upgrade to 2.2.3 (as you can see in the below GIF):
![semanticdropdown](https://cloud.githubusercontent.com/assets/5429033/18017086/77594158-6bd8-11e6-84cf-1f7933d9a443.gif)

In 2.2.1 I encountered this issue https://github.com/Semantic-Org/Semantic-UI/issues/4183 with the dropdown, which was indeed solved in 2.2.3, but now we have the above issue.
We really want and need to upgrade to 2.2.x, but we can't, because these issues break the production. 
I will appreciate if you give this issue extra care since it is a big regression and we don't want to stuck with an old version of Semantic.

Thanks.

**Comment by dorsha** [comment]
This change cause the "issue":
https://github.com/Semantic-Org/Semantic-UI/commit/33ea63ec9139ffd3cc49ae492aeffd1666f8db63#diff-dbb1b936fdad1f63b3a01f86d9a54fd1R961

Providing the `forceSelection: false` setting solve it.   

![image](https://cloud.githubusercontent.com/assets/5429033/18023652/680caace-6c04-11e6-8aa3-db4118e51098.png)

So it may not an issue at all.

**Comment by Ribesg** [comment]
Not an issue, rather a breaking change. I'm wondering why the default value of this setting is true. I don't see any use case for this, you really never want something to happen magically. The use didn't click, it should not do anything...

**Comment by ismay** [comment]
> I'm wondering why the default value of this setting is true. I don't see any use case for this, you really never want something to happen magically. The use didn't click, it should not do anything...

Exactly. Weird default. Also wouldn't expect this to be a patch version increase, since it's a breaking change. That's what lead to this mess. But anyway, issue's been resolved as far as I'm concerned, thx @dorsha !

**Comment by daltones** [comment]
Yeah, I after notice the option `forceSelection` set by default. Thanks to all responses and sorry for the confusion. But I also agree that this behavior is not very usable and confuses the user. I mean, if you are navigating some form and a random option gets selected, it really seems like a bug.

Even if this is the expected behavior, I had to patch `dropdown.js` to reach the way my UI needs to be (and I think this way could be more useful and common): It's like a "force selection or nothing” behavior: If your search text doesn't match any options or the dropdown was dismissed, search input will te **cleared** on lost focus (to not confuse the user that some new option was created).

Closing.
